containers/daap-create-schema/src/var/task/create_schema.py

```diff
@@ -0,0 +1,168 @@
+import json
+import os
+from http import HTTPMethod
+
+import boto3
+from data_platform_api_responses import (
+    response_status_200,
+    response_status_400,
+    response_status_403,
+)
+from data_platform_logging import DataPlatformLogger, s3_security_opts
+from data_platform_paths import DataProductConfig, get_latest_version, get_new_version
+from data_product_metadata import DataProductMetadata, DataProductSchema
+
+s3_client = boto3.client("s3")
+
+
+def s3_copy_folder_to_new_folder(
+    bucket, source_folder, latest_version, new_version, logger
+):
+    paginator = s3_client.get_paginator("list_objects_v2")
+    page_iterator = paginator.paginate(
+        Bucket=bucket,
+        Prefix=source_folder,
+    )
+    keys_to_copy = []
+    try:
```

Comment:
The  page_iterator will return a empty list if the paginator does not contain any items.so you can test using 

```
if not any(pages):
    print("The folder is empty.")
else: 
    Loop 
```

Dont need a try / expect here

Reply:
Doesn't seem to be the case, the page_iterator contains data even if a specified prefix does not exist so it fails with a `KeyError`